qiskit_addon_sqd/fermion.py

```diff
@@ -223,6 +222,13 @@ def optimize_orbitals(
         - Average orbital occupancy
 
     """
+    k_dim = int((1 + np.sqrt(1 + 8 * len(k_flat))) // 2)  # quadratic formula
+    num_orbitals = hcore.shape[0]
+    if k_dim != num_orbitals:
+        raise ValueError(
+            f"k_flat must specify the upper triangle of the transform matrix. k_flat length is {len(k_flat)}. "
+            f"Expected {(num_orbitals**2 - num_orbitals) // 2}."
+        )
```

Comment:
It looks like you're only using `k_dim` to validate the input. In that case I think it's preferable to avoid floating point arithmetic. Also, while we're here I think we should use consistent variable names throughout. I prefer `norb` to `num_orbitals`.
```suggestion
    num_orbitals = hcore.shape[0]
    num_params = (num_orbitals**2 - num_orbitals) // 2
    if len(k_flat) != num_params:
        raise ValueError(
            f"k_flat must specify the upper triangle of the transform matrix. k_flat length is {len(k_flat)}. "
            f"Expected {num_params}."
        )
```